### ISSUE TITLE
Albums with multiple album artists are split up

### ISSUE THREAD

**Issue by digimezzo** [digimezzo/dopamine-windows#31]
Currently Dopamine defines an album by the combination of album title and album artist.

This causes albums with multiple album artists to be split up.

Is there a better way of defining albums? By just album title?

**Comment by kbalint** [comment]
iTunes has a checkbox for the album page "compilation: album is a compilation of songs by different artists". Checking it will group all tracks, otherwise it will show as many covers in album view, as many different artists there are.

**Comment by Keith94** [comment]
>Is there a better way of defining albums? By just album title?

Definitely this, it should be displayed as a single album, otherwise it just looks wrong.

![dopamine_2017-01-02_01-21-48](https://cloud.githubusercontent.com/assets/5490615/21586800/d9872064-d089-11e6-98e4-ff68ccf13e1e.png)

**Comment by Riajyuu** [comment]
Agreed with @Keith94

**Comment by digimezzo** [comment]
I'm not sure I agree. What if 2 albums, from different artists, have the same title? I'm sure that can happen.
I don't know how other players define albums. That's what I'd like to know. 

To summarize my idea: Is there an optimal way, that handles all possible cases, to define an album. If yes: which, and is there an example of a player out there which does it this way?

@Keith94's problem is easily solved by using "Various artists" as album artist, while retaining individual track artists.

**Comment by Riajyuu** [comment]
I have an idea:
about groups dividing:
first divide groups to album artists. If same album artist, go for album title so album can be CD1 CD2 CD3, etc.
In that case, introduce disk code. Grouping should be done so far.
about sorting order:
first album artists, then album name, then CD code, track number.
(from foolbar2000)

**Comment by digimezzo** [comment]
I think we're over-complicating this problem. Itunes groups tracks into albums by putting all tracks which have the same album title and album artists together. Additionally there is a checkbox "Part of a compilation". I'm not sure yet by which magic tracks are then grouped together (maybe some custom ID or something like that). So, as far as I can see, Dopamine is not doing such a bad job, as it does just the same (except for the compilation checkbox).

It's time that users start using the tags for hat they are meant: album artist is for the overall album artist. If you fill it in correctly, there is no issue at all. The only thing I'm willing to add, is a "Compilation" checkbox. But that won't be universal (like in iTunes). As soon as you use these tracks in another player, they won't be seen as part of an album. Unless you fill in your tags correctly.

**Conclusion:** add "Part of a compilation" checkbox for those users that like incorrect tagging.

**Comment by kbalint** [comment]
just one thought: Apple has a database with all the info of nearly all albums in circulation, they can fine-tune this feature as they like. :)

**Comment by digimezzo** [comment]
@kbalint Indeed, Apple has more data than me to define albums.
Extra info: I just verified how Foobar2000 reacts when feeding it tracks which have the same album title but a different album artist: it splits the albums up the same way as Dopamine. So I can only conclude that Foobar2000 uses the same logic as Dopamine. Which seems the most logical to me.

**Comment by macmorning** [comment]
Actually I believe Dopamine already handles this just like it should. 
Set the album artist to "Various artists" or whatever for all the album tracks. Move the information to the "artist" tags. Dopamine will show one album and the artists of each track.

**Comment by digimezzo** [comment]
@macmorning I agree with you. My users however don't :) Mostly because they don't know how a program is supposed to find out that files belong to a same album.

**Comment by smarquespt** [comment]
@digimezzo 
I will also support the idea of having one album with all artists/tracks.

Clementine music player handles this correctly: https://github.com/clementine-player/Clementine

I don't know how they manage to do this but they do.

This is what it looks like:
![1](https://user-images.githubusercontent.com/1302668/33292996-1f3658fe-d3c3-11e7-947d-cca605793944.png)

Next image is shown with right click. The red mark - added by me- shows 2 options:
"Show in Various Artists" and  "Don't Show in various Artists". If one selects option 1 is obvious, if one selects option 2 the albums are shown as they are in dopamine.

![2](https://user-images.githubusercontent.com/1302668/33293103-862c77d2-d3c3-11e7-8ebf-9ff54e10d5c6.png)

Maybe this is a good option???

**Comment by digimezzo** [comment]
For Dopamine 2.0, I want to have a definitive solution for this. There are too many user questions and complaints.

I did some tests today. I took an album (= files which have the same album title and album artist) and I changed the album artist so that it is different for each of those files. 

Result: Dopamine splits up this album into multiple albums. As is being complained so much about.

Then, I took a bunch of other audio players, and pointed them to the folder containing those same audio files. I then verified how this album is displayed in the different players. This is the result:

- MusicBee: album is split up
- Foobar2000: album is split up
- Winyl: album is split up
- Windows Media Player: album is split up
- iTunes: album is split up

My conclusion is that Dopamine's logic is right, and shouldn't be changed. Organizing album based on just the album title would be wrong and will cause new issues. 

**Example**: Consider these 2 albums:

- Peter Gabriel: Up
- R.E.M.: Up

Both albums WILL be grouped into 1 album called "Up". That is NOT OK.

**Conclusion:** Dopamine will not work around incorrectly tagged files. Instead, Dopamine will regularly (and also just after indexing) scan its albums for duplicate album titles. If duplicates are found, a warning bar will be shown to the user. This bar will be in the same location as the update notification. The bar will tell the user that albums are possibly split up due to incorrect tags and will provide 2 options (configurable per album individually):

* Option 1: propose to provide a common album artist for the songs of the album. This option modifies the album artist tag in all affected songs.
* Option 2: show a check a box. Once checked, those songs will be forced into 1 album, without modifying tags. An internal album identifier will be user in the Dopamine database. This dosn't modifies the audio files. So next time a user starts with a clean Dopamine database, the same action needs to be performed again to group the affected files into 1 album.

**Comment by Keith94** [comment]
I like option 1. I've been changing album artist to "Various Artists" to fix that problem.

**Comment by jtc2002** [comment]
Are there still plans to add this to Dopamine 3.0?

**Comment by Jwn5k** [comment]
> Are there still plans to add this to Dopamine 3.0?
Bumping this.

It would be useful to have an option to select these split up albums (multiple from the same album cause of the artist names yada yada...) and just have dopamine artificially combine them. like a "select x amount of albums to combine into one". Maybe it would be similar in function to how Dopamine has the feature to specifically blacklist songs from playback that you can select, instead of removing them from an album/folder to not have Dopamine see it in the first place. If Dopamine can just index what file paths are selected to be "combined" artificially, and allow the end user to just name this "new album" made up of of all the combined ones (and maybe either find the album art or the user provide the album art as it's own option too), it would be a great addition, because having **16(!!!)** individual "albums" for the Hardspace Shipbreaker OST is quite annoying to look at, even if all the .flac files are all in the same folder.

**tl;dr**
Have a select/checkbox feature on albums and have Dopamine combine them artificially, just indexing selected albums to appear as one in the UI.

Probably restating other mumbo jumbo that has been said, but in any case, this is i guess just another QOL feature that would be very handy as it may be extremely prevalent to normal users and would be very great.
